### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by NadeemAfana** [Azure/azure-service-bus-go#186]
Not sure how to reproduce this one, but I am constantly seeing the following stack trace every few days:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x70d8ba]

goroutine 388 [running]:
github.com/Azure/go-amqp.(*Receiver).Receive(0x0, 0xe24b60, 0xc0002551c0, 0xd1a1f9, 0x1c, 0xe24b60)
        /go/pkg/mod/github.com/!azure/go-amqp@v0.12.7/client.go:1892 +0x3a
github.com/Azure/azure-service-bus-go.(*Receiver).listenForMessage(0xc0000c22c0, 0xe24b60, 0xc0002551c0, 0x0, 0x0, 0x0)
        /go/pkg/mod/github.com/!azure/azure-service-bus-go@v0.10.0/receiver.go:332 +0x107
github.com/Azure/azure-service-bus-go.(*Receiver).listenForMessages(0xc0000c22c0, 0xe24b60, 0xc0002551c0, 0xc0005a46c0)
        /go/pkg/mod/github.com/!azure/azure-service-bus-go@v0.10.0/receiver.go:284 +0xeb
created by github.com/Azure/azure-service-bus-go.(*Receiver).Listen
        /go/pkg/mod/github.com/!azure/azure-service-bus-go@v0.10.0/receiver.go:212 +0x134
```

```
$ go version
go version go1.12.9 linux/amd64
```
Azure-Service-Bus-Go: **v0.10.0**

**Comment by jhendrixMSFT** [comment]
I believe I have a fix for this.  If you're able to build using the changes in the PR it would be great to confirm it resolves your issue.

**Comment by jhendrixMSFT** [comment]
Fixed in v0.10.5
